client/landing/gutenboarding/hooks/use-detect-matching-anchor-site.ts

```diff
@@ -34,22 +33,23 @@ export default function useDetectMatchingAnchorSite(): boolean {
 		}
 
 		setIsLoading( true );
-		wpcom
-			.undocumented()
-			.getMatchingAnchorSite(
-				anchorFmPodcastId,
-				anchorFmEpisodeId,
-				anchorFmSpotifyUrl,
-				anchorFmSite,
-				anchorFmPost
-			)
+
+		// construct query object from entries that are not null
```

Comment:
If we're keeping the check a non-strict one, let's update this to reflect the actually check:

```suggestion
		// construct query object from entries that are not null or undefined
```